created fast inflow function

RickytheGuy opened the pull request:
The way I approached this is pretty different from your other code, so I made a new script instead

RickytheGuy committed:
created fasy inflow function

rileyhales reviewed (changes requested):
Looks like a good implementation that covers everything I was hoping for. Good use of pandas and groupby.

We don't need to keep the big array variable. You can place that read from xarray dataset statement inside the line where you init the data frame. Otherwise you keep that array in memory twice in 2 different variables. That bloats the memory requirement more than we need.

We may want to revise what metadata items we put in the netcdd.

RickytheGuy committed:
gitignore

RickytheGuy committed:
added 4-d / nondaily support

RickytheGuy commented:
I've updated the values to be accessed in in the dataframe init. I've added support for 4-d netcdfs, tested it and it works as expected. I also added quick support for non-daily files

RickytheGuy committed:
reduced num inputs, improved time

RickytheGuy committed:
updated

RickytheGuy committed:
memory checking

RickytheGuy commented:
Added memory check, simplified inputs

RickytheGuy committed:
updated function docs

rileyhales committed:
correct time data for rapid

rileyhales committed:
Merge branch 'geoglows:main' into main
